libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxUpcomingManeuverAdapter.kt

```diff
@@ -116,29 +93,19 @@ class MapboxUpcomingManeuverAdapter(
             val secondary = maneuver.secondary
             val totalStepDistance = maneuver.totalManeuverDistance
             if (secondary != null) {
-                viewBinding.secondaryManeuverText.render(
-                    ManeuverState.ManeuverSecondary.Show
-                )
-                viewBinding.secondaryManeuverText.render(
-                    ManeuverState.ManeuverSecondary.Instruction(secondary)
-                )
+                viewBinding.secondaryManeuverText.visibility = VISIBLE
+                viewBinding.secondaryManeuverText.render(secondary)
                 updateConstraintsToHaveSecondary()
-                viewBinding.primaryManeuverText.maxLines = 1
+                viewBinding.primaryManeuverText.isSingleLine = true
             } else {
                 updateConstraintsToOnlyPrimary()
-                viewBinding.secondaryManeuverText.render(
-                    ManeuverState.ManeuverSecondary.Hide
-                )
-                viewBinding.primaryManeuverText.maxLines = 2
+                viewBinding.secondaryManeuverText.visibility = GONE
+                viewBinding.primaryManeuverText.isSingleLine = false
             }
-            viewBinding.primaryManeuverText.render(
-                ManeuverState.ManeuverPrimary.Instruction(primary)
-            )
-            viewBinding.maneuverIcon.render(
-                ManeuverState.ManeuverPrimary.Instruction(primary)
-            )
+            viewBinding.primaryManeuverText.render(primary)
+            viewBinding.maneuverIcon.renderPrimaryTurnIcon(primary)
             viewBinding.stepDistance.render(
-                ManeuverState.TotalStepDistance(
+                StepDistance(
                     MapboxDistanceFormatter(DistanceFormatterOptions.Builder(context).build()),
                     totalStepDistance.totalDistance
                 )
```

Comment:
`bindUpcomingManeuver` is getting too long. Could we extract some of the logic into well-name `private` `fun`ctions?